twake/backend/node/src/services/documents/services/index.ts

```diff
@@ -514,18 +516,30 @@ export class DocumentsService {
         throw Error("Drive item doesn't exist");
       }
 
-      if (
-        !item.access_info.authorized_entities.find(entity =>
-          entityIdentityCheck(entity.id, context),
-        )
-      ) {
-        return false;
+      if (token) {
+        if (!item.access_info.public.token) {
+          return false;
+        }
+
+        const { token: itemToken, level: itemLevel } = item.access_info.public;
+
+        return itemLevel === level && itemToken === token;
       }
 
       if (
-        item.access_info.unauthorized_entities.find(entity =>
-          entityIdentityCheck(entity.id, context),
-        )
+        !item.access_info.entities.find(entity => {
+          if (entity.level !== level) return false;
+
+          if (entity.type === "user" && entity.id === context.user.id) {
+            return true;
+          }
+
+          if (entity.type === "company" && entity.id === context.company.id) {
```

Comment:
Here we will also check that the user is not a `guest` I did a similar function in this same file to test for admins. So user should be at least company member (admin, owner also working)